[DOC] Update Linux set-up documentation for newer release

Napsty opened the pull request:
<!--  Thanks for sending a pull request!  Before submitting:

1. Read our CONTRIBUTING.md guide
2. Rebase your PR if it gets out of sync with main
-->

**What this PR does**: Updates the ["Deploy on Linux"](https://grafana.com/docs/tempo/latest/setup/linux/) documentation with a more recent release, adjust suggested configuration accordingly. If using the configuration from the documentation on a newer Tempo release, Tempo will not start.

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

knylander-grafana commented on docs/sources/tempo/set-up-for-tracing/setup-tempo/deploy/locally/linux.md:
````diff
@@ -121,16 +124,15 @@ storage:
 ```
 
-> **Note:** In the above configuration, metrics generator is enabled to generate Prometheus metrics data from incoming trace spans. This is sent to a Prometheus remote write compatible metrics store at `http://prometheus:9090/api/v1/write` (in the `metrics_generator` configuration block). Ensure you change the relevant `url` parameter to your own Prometheus compatible storage instance, or disable ...
+> **Note:** The above configuration is compatible with the downloaded release of Tempo (2.8.x). The content of the configuration might be adjusted for earlier or future releases. In the configuration shown above, metrics generator is enabled to generate Prometheus metrics data from incoming trace spans. This is sent to a Prometheus remote write compatible metrics store at `http://prometheus:9090/ ...
````
```suggestion
{{< admonition type="note"
The above configuration is compatible with the downloaded release of Tempo. 
Future release may impact the configuration content. In the configuration shown above, the metrics-generator is enabled to generate Prometheus metrics data from incoming trace spans. This is sent to a Prometheus remote-write compatible metrics store at `http://prometheus:9090/api/v1/write` in the `metrics_generator` configuration block. 
Make sure you change the relevant `url` parameter to your own Prometheus compatible storage instance, or disable the metrics-generator by removing the `metrics_generators_processors` if you do not wish to generate span metrics.
{{< /admonition >}}
```

Napsty commented:
Hi @knylander-grafana 
I enjoyed a couple of days off, hence my slow response.
I've added the comment `configure total trace retention here` (helpful to understand the option) back into the config example.
At the same time I updated for the new 2.9.0 package. I successfully tested the example configuration with Tempo 2.9.0.

knylander-grafana commented:
> Hi @knylander-grafana I enjoyed a couple of days off, hence my slow response. I've added the comment `configure total trace retention here` (helpful to understand the option) back into the config example. At the same time I updated for the new 2.9.0 package. I successfully tested the example configuration with Tempo 2.9.0.

Thank you for updating and testing!
